### DISCUSSION TITLE
Overly-Agressive Auto-Completion

### DISCUSSION THREAD

**Discussion by Syntonica** [helix-editor/helix#5527] [General] [answered]
I only occasionally use auto-complete when I'm stuck on a symbol name or method name, so I turn it off.  I thought `<C-x>` would enable it just for the one instance, but apparently, it toggles it on and there's windows flying everywhere.  I'm not sure if this is a bug or a feature, but I'd like completion-on-demand as an option.

Also, another vote for trees. File trees, symbol trees, etc. Netrw-style is fine. Fuzzy search is fine for some use cases, but not all. 

Other than these two things, Helix is a very nice editor as it is. Well done!

**Comment by kirawi** [comment]
You can disable it in `editor.auto-complete`, and can bind a key to `completion` (ctrl-x by default). There's a PR for file trees: https://github.com/helix-editor/helix/pull/2377

**Reply by Syntonica** [reply]
First off, I forget Github likes to swallow anything remotely code-like.  I'm still getting my sea-legs here.  I fixed my OP.

I did indeed disable auto-complete and used `<C-x>`, but that just seems to turn it back on fully. I'm not Rusty at all, so I'm not wont to file a bug report unless I'm sure it's a bug.  If hx was written in a C-family language, I'd probably go fix it, bug or not.

**Reply by the-mikedavis** [reply, marked as answer]
There was a bug fixed after the 22.12 release about this: #5042. The `lsp.auto-complete` option turned off autocompletion from the idle timeout but not from triggers (typing a comma for example).

**Reply by Syntonica** [reply]
Thanks! That's great news. I'll wait for the next release.

(Note to self: check closed issues as well as open...)
